crates/misc/component-fuzz-util/src/lib.rs

```diff
@@ -773,34 +796,42 @@ pub struct TestCase {
 impl TestCase {
     /// Generate a `Declarations` for this `TestCase` which may be used to build a component to execute the case.
     pub fn declarations(&self) -> Declarations {
-        let mut types = String::new();
-        let name_counter = &mut 0;
-
-        let params = self
-            .params
-            .iter()
-            .map(|ty| {
-                let mut tmp = String::new();
-                write_component_type(ty, &mut tmp, name_counter, &mut types);
-                format!("(param {tmp})")
-            })
-            .collect::<Box<[_]>>()
-            .join(" ")
-            .into();
-
-        let result = {
-            let mut tmp = String::new();
-            write_component_type(&self.result, &mut tmp, name_counter, &mut types);
-            format!("(result {tmp})")
+        let mut builder = TypesBuilder {
+            next: 0,
+            worklist: Vec::new(),
+        };
```

Comment:
I like using `#[derive(Default)]` for this kind of simple case:

```suggestion
        let mut builder = TypesBuilder::default();
```